oauthproxy.go

```diff
@@ -490,6 +493,12 @@ func (p *OAuthProxy) GetRedirect(req *http.Request) (redirect string, err error)
 	}
 
 	redirect = req.Header.Get("X-Auth-Request-Redirect")
+	logger.Printf("X-Auth-Request-Redirect: %s", redirect)
+	// traefik does not send X-Auth-Request-Redirect, so give the traefik headers a try in case X-Auth-Request-Redirect is not set
+	if !p.IsValidRedirect(redirect) {
+		redirect = req.Header.Get("X-Forwarded-Proto") + "://" + req.Header.Get("X-Forwarded-Host") + req.Header.Get("X-Forwarded-Uri")
```

Comment:
Take a look at this PR, it added a unified way for `X-Forwarded-Host` vs `Host` usage.  Let's use that helper here

Reply:
> Take a look at this PR

What PR are you talking about?

Reply:
Is there a link missing? Not sure what helper you're pointing to.

Reply:
Link is definitely missing 😅  https://github.com/oauth2-proxy/oauth2-proxy/pull/729

`util.GetRequestHost` is the function we pretty much aligned on everywhere instead of `req.Host`